packages/core/__tests__/custom-cache.js

```diff
@@ -111,3 +111,56 @@ test('emotion plugins - @styled-system/css', () => {
 />
 `)
 })
+
+test('why this exact api would cause a problem', () => {
+  let cache = createCache({
+    emotionPlugins: [styledSystemCss]
+  })
+
+  let SomeComponentFromNpm = props => {
+    // let's say i'm an author of a library using emotion
+    // i want to apply padding: 2px to my element and that's it
+    return <p css={{ padding: 2 }}>i want this to have padding: 2px</p>
+  }
+  // as a consumer of the above component, i don't want to have to care that it uses emotion
```

Comment:
@mitchellhamilton how is this problem conceptually different from stylis plugins though? this still is some kind of contextual processing in both cases

Reply:
It isn't conceptually or technically different but it's very different practically. Stylis plugins generally do things like RTL, add syntax or etc. with the important constraint that it's less likely to remove/change how syntax is interpreted.

Reply:
Even with RTL plugin, this can break stuff somewhat easily if a component doesn't know about being wrapped in - because not everything should actually be flipped to the other side and using `/* @noflip */` is necessary in such situations - but the component in question doesn't handle it properly, because it might not be rtl-aware.

Reply:
Yeah, that's true. I think the main distinction is that stylis plugins are a much more low level thing which not a huge number of people will use whereas this is intended to be used much more widely.

Also, what this is operating on would break a lot of our current assumptions which would mean we couldn't make some of the optimizations we currently do. For example, we assume that `css({ color: 'green' })` is equivalent to `css('color:green;')`, this would mean that that isn't necessarily true.